src/goto-instrument/contracts/dynamic-frames/dfcc_instrument.h

```diff
@@ -372,6 +422,51 @@ class dfcc_instrumentt
     goto_programt::targett &target,
     goto_programt &goto_program,
     cfg_infot &cfg_info);
+
+  /// Builds a dependency graph representing the loop nesting structure.
+  void compute_loop_nesting_graph(
+    const irep_idt &function_id,
+    grapht<loop_graph_nodet> &loop_nesting_graph,
+    goto_programt &function_body);
+
+  /// Tags program instructions with pragmas
+  void tag_loop_instruction(
```

Comment:
can we build the loop dep graph and tag instructions in single atomic operation ? Just to make sure there is no possibility to use outdated loop descriptors to do the tagging.